### PR TITLE
55 slu status statuses endpoint

### PR DESCRIPTION
**Pull request by jacekroszkowiakdev**
Work on #55 finished, please review and comment

### CODE REVIEW

File: one-shot-device/src/devices/device-registration.service.ts [hunk 1]
```diff
@@ -141,7 +141,7 @@ export class DeviceRegistrationService {
 	async getRegisteredDevice(nonce: string): Promise<DeviceRegistration> {
 		const device = await this.deviceRegistrationModel.findOneAndDelete({ nonce }).exec();
 
-		if (device == null) {
+		if (device === null) {
```
Comment (mastrogiovanni) [inline one-shot-device/src/devices/device-registration.service.ts:144]: Please change it with
```
if (!device) {
```

File: slu-status/src/slu-status/slu-status.controller.spec.ts [hunk 2]
```diff
@@ -73,6 +90,13 @@ describe('StatusController', () => {
 		expect(result).toBe(response.status);
 	});
 
+	it('should return SluStatuses', async () => {
+		jest.spyOn(service, 'getStatuses').mockResolvedValue(devices);
+		const result = await controller.getStatuses(ids);
+		console.log(result);
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.controller.spec.ts:96]: Please remove logs from tests

Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.controller.spec.ts:96]: Please remove logs from tests

File: slu-status/src/slu-status/slu-status.service.spec.ts [hunk 3]
```diff
@@ -71,15 +76,18 @@ describe('StatusService', () => {
...
-		const result = await service.updateSluStatus(update.id, update.status);
-		expect(result).toBeNull();
+		try {
+			await service.updateSluStatus(update.id, update.status);
+		} catch (err) {
+			expect(err.message).toBe('Could not find document in the collection.');
+		}
 	});
 
 	it('should return status', async () => {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.spec.ts:93]: Maybe this line is missing?
```
const getSavedSluStatus = await service.getSluStatus('...');
```
Where is defined `getSavedSluStatus`?

File: slu-status/src/slu-status/slu-status.service.ts [hunk 4]
```diff
@@ -9,16 +9,46 @@ import { Status } from './model/Status';
...
+
 	async saveSluStatus(sluStatus: SluStatusDto): Promise<SluStatusDocument> {
 		return await new this.sluStatusModel(sluStatus).save();
 	}
 
 	async updateSluStatus(id: string, status: Status): Promise<SluStatusDocument> {
-		return await this.sluStatusModel.findOneAndUpdate({ id }, { status }, { new: true, fields: { _id: 0 } });
+		const updateStatus = await this.sluStatusModel.findOneAndUpdate({ id }, { status }, { new: true, fields: { _id: 0 } });
+
+		if (updateStatus === null) {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.ts:21]: Please change this line with:
```
if (!updateStatus) {
```

File: slu-status/src/slu-status/slu-status.service.ts [hunk 5]
```diff
@@ -9,16 +9,46 @@ import { Status } from './model/Status';
...
+			throw new HttpException('Could not find document in the collection.', HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+
+		return updateStatus;
 	}
 
 	async getSluStatus(id: string): Promise<string> {
 		const slu = await this.sluStatusModel.findOne({ id }).lean();
+
+		if (slu === null) {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.ts:32]: Change this line with:
```
if (!slu) {
```

File: slu-status/src/slu-status/slu-status.service.ts [hunk 6]
```diff
@@ -9,16 +9,46 @@ import { Status } from './model/Status';
...
 	}
+
+	async getStatuses(ids: string[]): Promise<any[]> {
+		const devicesStatuses = await this.sluStatusModel.find({
+			id: {
+				$in: ids
+			}
+		});
+
+		if (devicesStatuses === null) {
```
Comment (mastrogiovanni) [inline slu-status/src/slu-status/slu-status.service.ts:47]: Change this line with:
```
if (!devicesStatuses) {
```

File: slu-status/test/app.e2e-spec.ts [hunk 7]
```diff
@@ -71,8 +90,8 @@ describe('AppController (e2e)', () => {
 			const { status, body } = await request(app.getHttpServer())
 				.get('/api/v1/status/did:iota:1234')
 				.set('X-API-KEY', '2b3fe07d-b7db-49cb-8300-d32139e3d435');
-			expect(status).toBe(200);
-			expect(body).toEqual({});
+			expect(status).toBe(500);
+			expect(body.message).toBe('Could not find document in the collection.');
 		});
 
 		it('/:id (GET): wrong api-key', async () => {
```
Comment (mastrogiovanni) [inline slu-status/test/app.e2e-spec.ts:97]: Is there some code missing? Where body is defined?

**Review by mastrogiovanni — CHANGES_REQUESTED** [review]

File: one-shot-device/src/devices/device-registration.service.ts (hunk 1 above)
Comment (jacekroszkowiakdev) [reply one-shot-device/src/devices/device-registration.service.ts:144]: Changed.

File: slu-status/src/slu-status/slu-status.controller.spec.ts (hunk 2 above)
Comment (jacekroszkowiakdev) [reply slu-status/src/slu-status/slu-status.controller.spec.ts:96]: Removed console.log

Comment (jacekroszkowiakdev) [reply slu-status/src/slu-status/slu-status.controller.spec.ts:96]: Removed console.log

File: slu-status/src/slu-status/slu-status.service.ts (hunk 4 above)
Comment (jacekroszkowiakdev) [reply slu-status/src/slu-status/slu-status.service.ts:21]: Changed.

File: slu-status/src/slu-status/slu-status.service.ts (hunk 5 above)
Comment (jacekroszkowiakdev) [reply slu-status/src/slu-status/slu-status.service.ts:32]: Changed.

File: slu-status/src/slu-status/slu-status.service.ts (hunk 6 above)
Comment (jacekroszkowiakdev) [reply slu-status/src/slu-status/slu-status.service.ts:47]: Changed.

File: slu-status/src/slu-status/slu-status.service.spec.ts (hunk 3 above)
Comment (jacekroszkowiakdev) [reply slu-status/src/slu-status/slu-status.service.spec.ts:93]: I don't understand. The line is present on local.

File: slu-status/test/app.e2e-spec.ts (hunk 7 above)
Comment (jacekroszkowiakdev) [reply slu-status/test/app.e2e-spec.ts:97]: body is defined in line 131:
 const { status, body } = await request(app.getHttpServer())

**Pull request merged**